linux/DtaDevLinuxSata.cpp

```diff
@@ -487,7 +602,10 @@ void DtaDevLinuxSata::identify_SAS(OPAL_DiskInfo *disk_info)
     }
 
     // fill out disk info fields
-    safecopy(disk_info->serialNum, sizeof(disk_info->serialNum), resp->m_T10VendorId, sizeof(resp->m_T10VendorId));
+    if (!g_compat_bsd)
+        safecopy(disk_info->serialNum, sizeof(disk_info->serialNum), resp->m_T10VendorId, sizeof(resp->m_T10VendorId));
+    else
+        safecopy(disk_info->serialNum, sizeof(disk_info->serialNum), resp->m_DriverSerialNumber, sizeof(resp->m_DriverSerialNumber));
```

Comment:
I wonder how do you expect `m_DriverSerialNumber` appear here?  SCSI reports serial numbers in separate VPD page. I guess that is why `sedutil` on Linux cheats with using `m_T10VendorId` instead.

Reply:
Ah, I guess overlooked that in specs. Updated to read from VPD 0x80 page.